packages/app/src/pages/transaction/index.tsx

```diff
@@ -94,6 +100,38 @@ export const Transaction: React.FC = () => {
     return account;
   };
 
+  const setupLocalNonce = async () => {
+    const gaiaConfig = await wallet.createGaiaConfig(gaiaUrl);
+    const walletConfig = await wallet.getOrCreateConfig({ gaiaConfig, skipUpload: true });
+    setGaiaConfig(gaiaConfig);
+    setWalletConfig(walletConfig);
+    return walletConfig.lastTx;
+  };
+
+  const getNonce = async (): Promise<number> => {
```

Comment:
This looks clean and will work in the vast majority of cases. I just implemented a fix for the same problem in the faucet, but using an (admittedly brute-force-y looking) stateless approach: https://github.com/blockstack/stacks-blockchain-api/pull/272

I couldn't see any other way to avoid a `ConflictingNonceInMempool` error when another application's local state uses up `lastNonce + 1`. For example, multiple sidecar deployments would trample over each other's local nonce tracking. Just a heads up in case you run into a similar situation.